### ISSUE TITLE
Using sql helper in promise chains causes unexpected preventAwait errors

### ISSUE THREAD

**Issue by icopp** [kysely-org/kysely#693]
Here's a simple example:

```ts
import { sql } from 'kysely'

// a plain promise for the example, but this could be a value determined inside a transaction
// or as part of some other promise chain
const someTimestampValue = Promise.resolve(true).then(
  (value) => value
    ? sql<string>`'infinity'::timestamptz`
    : sql<string>`'-infinity'::timestamptz`
)
```

This results in a raised error: 

```
Error: don't await RawBuilder instances directly. To execute the query you need to call `execute`
```

This is non-obvious behavior for a tag function, where I think the expected behavior here should be to act much more like a generic object (e.g. `const a = b` and `const a = await b` should have effectively the same behavior).

**Comment by koskimas** [comment]
If we remove the check, we get a new issue: especially poeople coming from `knex` will be confused when

```ts
await sql`some sql`
```

doesn't execute the query. Since you are the first one to bring this up, I'm assuming that'd be a bigger problem.

You can wrap the `sql` snippets in objects as a workaround

```ts
const someTimestampValue = Promise.resolve(true).then(
  (value) => value
    ? { sql: sql<string>`'infinity'::timestamptz` }
    : { sql: sql<string>`'-infinity'::timestamptz` }
)
```

**Comment by marceloverdijk** [comment]
I'm hitting this as well to use a `async` function to enhance a query...

I'm trying to create a function that enhances the query (`SelectQueryBuilder`) with pagination, sorting and filtering options that are derived from a HTTP request.

This function looks like:

```
  async applyQueryParams<DB, TB extends keyof DB, O>(qb: SelectQueryBuilder<DB, TB, O>): Promise<SelectQueryBuilder<DB, TB, O>> {
    const data = await this.getValidatedData<PaginationSchema & SortSchema & FilterSchema>();
    const page = .. // derive page
    const pageSize = .. // derive page size
    const sort = .. // derive sort
    const filter = .. // derive filter

    console.log("Query builder before pagination:", qb);

    qb = qb.limit(pageSize);
    qb = qb.offset((page - 1) * pageSize);

    console.log("Query builder after pagination:", qb);

    return qb;
  }
```

this function needs to async as the getValidatedData() is also async.

in a handler I call applyQueryParams like:

```
async handle(c: C) {

    let qb = c.var.mydb
      .selectFrom('customer')
      .selectAll('customer');

    console.log("Before applying query params:", qb);

    qb = await this.applyQueryParams(qb);

    console.log("After applying query params:", qb);

    const customers = await qb.execute();
```

but this fails with:

```
Before applying query params: SelectQueryBuilderImpl {}
Query builder before pagination: SelectQueryBuilderImpl {}
Query builder after pagination: SelectQueryBuilderImpl {}
✘ [ERROR] Error: don't await SelectQueryBuilder instances directly. To execute the query you need to call `execute` or `executeTakeFirst`.
```

I understand the rationale behind the `preventAwait` for knex users, but Typescript users - not coming from Knex - are now confused and limited.

**Comment by igalklebanov** [comment]
This'll be resolved in v0.28 following merge of #1160.
